### ISSUE TITLE
Why is using DES decryption twice slower than JS

### ISSUE THREAD

**Issue by Zz1056627814** [originjs/crypto-js-wasm#11]
I tried using cryptom in webworker

loading completed first, taking 20ms
```
import CryptoJS from  "./cryptom-js/index";
```

```
await CryptoJS.DES.loadWasm();
```

performing decryption operations does not result in better performance compared to JavaScript

```
function getDesDecryptString(encrypted, _key, _iv) {
  let key = CryptoJS.enc.Utf8.parse(_key);
  let iv = CryptoJS.enc.Utf8.parse(_iv);
  let decrypted = CryptoJS.DES.decrypt(encrypted, key, {
    iv: iv,
    mode: CryptoJS.mode.CBC,
    padding: CryptoJS.pad.Pkcs7,
  });
  return decrypted.toString(CryptoJS.enc.Utf8);
}
```

this is my test encryption code (DES,key:'EGCNPCV1',iv:'CNPCV1EG')

```
1DuWAPUp7h+602P/nsCSR9n5xh8qVOwkXclJe0rMRp2q0SZ9izHCcwlMg+HR2dBexrYTfeqkcO0n2w7fSQCSdego5eJAbY8XxK09HlEXJQYEHUCHK5LmW/GE63itMZ5ZFVaffLr2SiIlCOSeaJ4u0nR/J2E2LQMmh9PQzrRsNJiaGLUrP10kt3xFMohoCkMKh4KEFDnYlb6udO2blmqqBaDpP4DBlPXJxnufli0XWPTrSh318uk6cdLXhLUxRVsLCtUqZykNVd1hwNHs4XUPjRfVj1QpgihY4jBF7X6CmLsGPqcdrmOXSHvzFO5/dYM9XQ6jbMm79WgMJUe0vgbVJdpcxwypej5WJDkyBoV+SlDJVCzLEpW++moF1KVXdwJSG1sNIzUw4sGzjy18mE5Fy06BtfrO1YF8PQzq4DTByWt57yLIBVTHuB3tPRi4AryHfTD09Gk13oAgn7zZALYyCgCsyVs8HMz3pqD59WM3wNHtMm50vIJejTitwYQvZ2dwZsRG4DrRnTJYrCuF/oR2HktwnOtRaSfOC1ShlqAtLjwdF5fWN4o0XemWdGg80v8fNcUjWoJbG2qXaOtpfzhC5yOYM+qkt+YG05KTuEDmbROy6+AWTlhooYrivjQcYFrYadO63tMZEAMcgzcaxzgci1GgnrvTqbHI/84pS/OI5p+RLAA+UkD35KNbM7kGofX9P6fcHgxvTZ3hbhZd/fMV4WJYJUwIcUbzPfVqM2Xl223w1aaoFCcmPyAsVLGmmZM/WvNds7UUY8gDk4qcmUOXv2jxpWGK0SnTYtKygQXcTxOKRYS1/ZH3QthvAFRcOuSUKktu7+V/JNsQ2IbfSgSm7RKHRWkFnD2NylTXMXwOsnXzaG2hZp8dtD5BvmUu4vXj/9S+1djseZrtqjbnP/PP3JzKGey37tOUR1emfszsuV9lreARlXm0gEExlkFy1kgRhhElg0Ta30sVKqa4NeoE7oViGkcAdwMHVgfb1pRZQxo7ntpkqUCo0KBjlLGAczivw4ofqSB075KunKccqpjKtFrDHHd7jnCF1S5Sbej4W0/rrv4dQ7C3R+Pr61W/o3VRKbJJFSZ2DM/ZyDAqutKegPLiWdNTyfHC3JZ5XpSNPBUYbeDYP5BaeZoVK1xCOSrexro00z+mnD3YR3xnFNU4Mp+OAAaXBDJRhAbmdpGczkAIlcvcq48KCfaUMgNJaT/gROtuzG8jdaomIYRrOxVtFj2Qr8ed891igq4nTyVc2iLHUN/jY4LoEwB+5NVqagD9aKubO7R2Dz/ENaL4ATV4bvv+oQjyNV6SSrgv1s+2EPBIHKLnskR0I4mraFfAc1YEBXeV/1HXLNFSfMFtfUnmtwTNiwc1t+4Jo7PM1e93XGI9dTHXog4olwfgfHB40pDpLLT1pchPbQZ+swR2RsZLcqwnSzaoK49hhPfqROIfaCWWvoBq4LD4vKxccNSmieMBRqWMhthTzMGuvGoKtPHSOqPWVS/pn3rINfc2ZqOHEV3o75x9SnURvhFOkpjsRpr8V01bXu1hy8Ll1K2fNyuhMT8NTULQ+MzBxJwUMjQlP/+fNVD5KWRUCBos13cXuXYM8VYLQDut1uFWQfJFy0IVzVMWh1UUZMHcTSJrCOGmIRoXXropGFJx4cUPydZ92CS5GhIHx7p9VWT6a0/s+zhEZiMmu+KwJpvfDESwOKism0jhvJbnIJwWChgLEJCOKBTFPHS84SJFlElD6ZoUSO4X+ozx5a78nFfoRS76+7OzooQE1AXf6O93zq/x3HWdlJIHsWO3zFUNKGFYH/0pbBh+vvKjduoEiVl/sahaMgEGqni8LPQEEsJ+c5/Jj55kqLIR4YNabcIYY9iRHlAaXuNaJMr3OYUk4erqlEXyHJIC94ulwFKb9nx7aBjqxl82u5xe1ynojOUvU4eYlf30AZtoQJrA3FDwkk6R5LuhjaQExpQQhoN/bntvlrT+LuyGN66c2kSmT0sKI08ajYLY2A4F/2fEpjKJUhGZufZMqs2YQv5DNOmiHOqjDDPVj98/9g/3IStxQ6Igjwk26z8lz8/WO3Fz5BW42gFbcmebZfauZLawOqgkotmZ2OjnmPAgUQCgcQBvetgI9uirg2bZJMrMjMbto2v0XKBjFvBMRfW4spRnMih2tGZSiXWtHBpdG/Pj7JIhPe/rlWiKUdAdMxo/61uNTbtnPrzhk0dz++PpweXzkioosCrWdGq7SlTEEqnar2l/xcupezuTkRp2EW/vBWqrliFLMxQXS/q+T0K3iWd/iQ8yDj6PBuU7vfsn1D9znoT+XxyetC5sXYpBT6hTcX/M/taj0RquzsAVVQwK1g6jIMPJWqIGBIddIKCQ22b93OBxURf/mL+Z0mqP2ULJ2S/aQnU2TaVa2OKaSicVrNxobUT9LSStZW3tGjpBRGOWIw0gAxWbG3K5/njkBaKSUCmCk0aPPGb2gTK1lS/U18XaKqBBEaJCN6uEwfFA0cfeSdXhlQwFOL9/I/bH+/flNZPDgeQu1jZLuVm2/kt0/DkNsB24s7WlJ+NFGb20kJXGxxVOHPaFP3yvvN/8iQBzdFv54gIZ7aqqfClBbf12O9ycpow5hgkZu0wIBduh+gtlmANaxJW5cxYfe2IsxhWudAItMluz1R4nV7NrXSMc2wXN7V2JoG6CXxmGLLlSO6PsShJ4NB94vnlaVn8lDIJD/Sizss+1urMeT2JyUOwMVZJUAluK+2mVdY8FCj+92u3vjdMQajt+fd+7b9B/gJ8644QhqdYd70yFADP+EfDQY1kOff2I0X5z8CwDD0upnlCaMrizVrmzjQvLXcDr4Kab/UxzmbSG1nd9nVBjKyVeOVMVibRYDCExAiXBwHwGQafX/MmWoDsz8Z2V2LrWUiFEM3mlIZgyo4doTU5Su1xcEzPL5s1tK34i/c2vmpmWqdFffUOTyy8IHxnEFIIZWfhW6j8dpq50NouduU5WPDCDvTmrRkDDN9fSi3QY6Y6YdUBN/ECqnuhTgre3a6zmS5/N70h5CqasPbu+BXWVUMbNamBCqHmwa/fVKvJa5EOOGB4qEiSH7S4pT8cjSGD74dUCjGcHEp+GgKkWCIzzT/WDN9N0OpgLbqbDNcvvLXGRz61hZzypIcwmTT1zbd5bCU7YwVHNIA8MFkESl8ptH/lGI7YuFMCgxsOTWbqrcmqxuOkVgu7eBgOY/3Z9MNZTG8ebd5YGhX8XWaF4Co/Dwb0bebMQS65kPUxqpzHAUyZKCiR71kNcmTGHS5aRUDd36Ni1hHupaaTjZt/k79e5B6laVv15NjJ1WxZC9nIGXMLYff6C4MX+5HqibO2OKBJNPB0eWWrYftsjjU7rONuqtftpQlQXx9B77vWROE/kIhjIBKvCx4SrqFJcojHN3jD31wDz5htj6MRyAPS2sMyUOH1soyVzHrGPiuvOa7mwW62D9UqRUT+ajWVnODhVtxnEtM1DkANQRkGswPz3S5KD0Jlq4JodB8j0dmyfuCgKkm14JjOP98alNqw70k610yE2ja+9XTYwZ4I+qsznYkh6fKKY1DshuN2n08IVnw7AshIN1nEncRGfoFR3sZKBQsE2InhLauFgfSYqpklDJ1NZBW1hVRWie2DeJwPua0EnZg4rNvXE1LD07W1Wc+KGMTYjtdyeSWhnN7BtK++EOq9sD2CWW13asoLdy7Cn2WRr59DfCSCzlzhN3h+p+eTIprXy6tZmiymkwGEYzTtdIDd8B/KU3XmTFMoDXLrsxemJ9/lmsm0CkmuWdqYUJwybhiTBKAeVhyzf4sOcHcV7Wo7LzOtj7fBIMG+2irM8+d6ieNVpeYrWO1MEaKV8I9gcrbUY0vxBFHRguL/6pJ7wvCNfRyfCpXL7LJdmHCMYUoKODYhMQYcyjRCpA10I5G+3EcIuP7CWvmRuOu2yfbxLfl88+HJGrnpYzIPd2j+mKyTdRXlSJkxFRi8PFLWRNRbaB9RP4nFr/FKzrQ5au4w2RMwIL4xF5hV7ggdNPOQkDBhpQfxxL5SzyH11zZGNvWoMw0Dud48eH9g/9bgYOYlm89rwO0qY57CZscV5vrI9SSel3fShB2U4giSoJZ0JypZeofpwfLveIczf1zliiIJ/kSGiqve9aFwcDgvQpFhfeyVy7z131DLnvQ7MyrEWHY3S6AcfJ3yjAB5QSW1fCzL8gCtFBrRzEPS9oyXJmwACrLp9YXlBoQl29a4deWimNvVUozwiAsmMIiqi5twSLJKtDwWOQTylTMVNLQNFAtX9QAllgCKgd6DO3AlYngJhNv9gqpHqdLfWi0QlmFQiVSpu+ROCCtaYZrl02qXR7tpz8tdjcBxjoq6qtBuOTttcyB7LAHqyth1IvxNvFAvxDjKfzGuwpPZQW+XTGHpnSnLRfg2iUtkc5CYCg6WkBimRJgNviGoqJzEbzSYYklMcCMz5BGjX2Hn5eCfhdoGbrtYvHHg97/qwBcGAnriJDFO6AZgE/MLqqY/FDgjdcZAblVi1G7M/Ss+yd/1fNJPLxPx67PqQUvpHv5WKz7efCCaZjXOKqOM6n6U4YjJBmd7BhZ+jDh4QnQgsqS1NdNVS0fozlnOztdzocW1y18dPxp2zFlgUDL5VuAJ36WtMDxS0B+tE5ZX126MGCqITXJRRAzOYvINWhKDBGjKV/G/xZA54OvxLD+OJCEp6OGD9oldNGadXEP/L/UUf1Q+U3R0FbaHJoUVXeqaQ+li+8z8XXPzUx11A6UHuGthO0dPvNNhqW22HRC8naTRfeqs2X60EqJ9lFt8uboEGNkN0cTMiU0ukk6CDlYWG7YTz8TW22BdN/Bi5B4U47iateoj+RH0A2YebrXBoTaKWG3J+++F4zKPi76u50TFFYb1SwyrwNGQS005OmKkuq/lsn2+UoIVwWzR9Jl0ZVhXCjStQ8bWtQI1Zxtnx69eQl5T1RECzD57Jewkcxc556OOapewH2rSQmiVR0hjKs9hf7MYl+z81LbD6Thkdd9iPEjqb6sVE21uvlFiaAMNk5dlz2Pzm46+tsf3EA+pm/fT8t0HHrBpzZi9ILiknF3u6UJXHmBTv4Eg8TLhhwswJjSuTyysnMp+tWDPCZn0ynaJ2Oq0R6YK2gc1KbN+3RshhJi0iz0tKLOHsbXjURPJSUOJQfw3pYthuZntpXL+pbwAgmJPPuCll9nn8KsdCjZossGOZjTip9K54c6a9DEPsiFx7tZIBJ5cc5rmH8TQWHoqlUFSLQBonAb0vwIXGP+Mdg0B+A5aj7pKWRDtjofSrr2CKg+G1flk5vLqRQj10XfOuYEQlqyuRuCfvC3Qm+GQZhkEX9/tEbS/rON7knVw/BE920vWpRY0uZL2eXiouo+QbN+8A4wNdNmlHCFDR3ITjwEifJytJW9IXanPJ+Du9MpYKBKg+9syIyGo5+ESHHcdtF+zn3RkRXKSdMLZ0mS+YlbZjpYD4TMQIy0J0iU0vL34H7ZrRdQvTAOpvjeKDy/egi1F0/NSWknxh0YFXaWZRH/Reu0D/f1OFEYjqdLrE6Drb5D8HUekYNmby32mwJp8ehDk/Sp7bbf5jM7w2DjxJWRHvnmV4lH7k7WWw+tMrxbpF29hxw/FhODVg3TR4dwg7GZa8tYhsRyglLbPc92YU+Ebo5qri8T4zgpaqyO593GesGCUnXDEFN9fWdTscxOOm/uXN3ckOXth0yr8/zynkU8oVWxGHA2ACtGoo9mELJRQpy0YVAMzv/HOZl5aCi0xIW+Ro1lwjv0rOiuq5GIsByIUXWdfx10lPU1iqqiyHFmQKKSkY5V92ckpmw2nfIMAVsw9hdxC0xa6VNQJPubMyZ1TsIYLkv8wTdwpYnQeH7ISRP9jPCO7zZp4OOlDbp3788Vjb5ykfzj8gYwbjZGmH8eE7IJgzfWpnAVms35aq2c8YvBABG77+jowhfTIrrP0k+XOwXG2f4jlkcANzPj04U5Z90qzYZTF4iuO1YuQBNA06HKfRXvypMMCurkgi2InivzihWzLiRjs0yFoWWMxAIHF+S5ChJgIVLMF+YFC0AB0iROKHSuCVQOdzSiWBtQKU9pXoxwv9wc6kE4tAWR5gDjEjdTnGQMtDAdf1Z19U9RsnMfiMWFhNybQcxxvXplXkDIe6jJSqNNlkDTY5gh6vHWzguWVpDf07cktQBC6vN5C/tF9tsQGtcs0+FRqT7NjmoLTw7zA0MpE9c3EeumymXiqjXIwQgDnPJasf748dvGFfkRVzM3PtyTQiv6UYicxpB9YoJjMLyuelIcRXeeGfDGQJLHps84EouAvMJp3vaBuPHLp2Od/4Pz5y8AmfSnSAv49STRMsFGF9sj/e19oM5IxQe85S+NVT7atGKUKrxJfvpaeUyAEosMD95UnFUR4aA1AiDiy/cI3SCkCYZHsN4VrrN2o7HoXpjGshb7YA+qnLTxDZSp7lK9KqnKfTW4dJGcY4GXEq+L0ZaqO8bzP8dM4/OjNpZoQ74jxb6iJv70Xz1HINrjdxY8rAC1KusNyWx5g29TbbPMvp2B6vS5JNz95yaohlUH7d3iTVXqnkw2o1dZhkKbaDPjT+Ef52qxWqwTOAQGI4kOuqAzY4lG2P1rus2wxTiU8UQJdCmPU9VqE+ukm+BEGqXvzq1Wruw6Z9hyi1gj4Gkao7KAdg2zHvuHlP6qSYxJYpaVJZp7oTEXlMXDBvl7of471sUtu1JOu0a2tOLj0UdQqrb8qA+8omcvV5aQ4tNOFbIZcrwwognK6ssfvC6YU2obG6nZOZCkMA6/F4pLVJkZzynYJ/e+QS1coy+ltNIPjHrCTWGZKxzCmm1PN84vBBOv8uyZPlW7V1PCny7H3wcFR8JmLPpwnfCvb4lTx0fE3VFN9xNEa55o8tbjRe3NgJfrFTPQiR7I9eJd8ibixR8llkj+PgAP1PiCQrOrDpD5B9FrSoLDYM2ndN1DtNDrukUjiGYL9QdKomBbFaE+6jJdrRnpLJprHNv6HPve/koZU0zHTMMp8e8uGr6bjSjuX5KjIzJrPSc2NBDjtq9kaWyNBJ5Okeu2jzQz+xgqpuk72NzzbiMx+lCFH5torE7uI6ufXdOpMZAm+VfZ5lrdAUDj7oRgvRdbSRjLYUTTO0K8b0prI8pmc2gElZ1hF55dQA3qt3WoKiK4UZ3LenBczyvVuKmqwlIWCLyUUtoAHx359uiYdxHXvUD0++kT1THGPSn1zdgLhpNnjlodp4Y1ly42QGMFJO/Qo0UF0fXI64zWsPKT1QRycZOQb2xBnm09mhpuezPgyKZaKoFOX18a4Ud55deFJcpaIpvvAGKNm0vy5CPVNRNwdXhC6JczHtbQK0TwRJafZiVJQPEbLkrGRWYo0/IjYBBhVjKVILBLHPDgdmK569rGIskamqt+PR58l68z2ZEgs0NeIe6ayt7JPSm4YeLrk2z4cZgAF+kiJ3bgFciIdA9NrJbXkX/qEmX7Oj4MjBFr7utKRiPq2CT+OEjnJTmfTtutvp50YiNi0brerFjDtP0HC2jetJnhO8ThRTMLstOlQn+aYkVmzheA2y2VojHXwPOwKaSEqoBpM0uFGPLRRCNIEjESeGYLjiylJjB17ctkvlvn91szfMLZss7BtXMqC20z6oMGnPymonX3VkW3tdVMzfZN2oY5v6SbuELW9eAEZu8gxY0vuQMmgyQfbENOSsXj2VfWrqMl8fpNsP8pL5Iep3r/cuz1+3ZnkISgeGef2vNOjBm2Beub3a6HzNvJy5B4hn7bqTllxgv0dWpX7wHaE9RxEuiwgc7GLEETQQVrZEPJ/i1hPukaCM9N9S6Jkfb4jzbQ+cn0ZSO+jOBrRYdB2qGdQmpSKaHFJH3UXVr1gYvST/Jrpc2TQyrZNxp47YOgHEJH8n9NAmkRcSRF9vSIYHMdmuHx0htQ883+j6sSMWp79cNdRxJl8cvw7U/fpTepsoUZ6XhclnkQMfWbdxyLDmqAjAuL+qXNwyKay1+1persh1QTian82yS+Ik8tn8/L5fIa+I1WT313Yn7KFAsJZ3uJ0Hu8waoKIWSjNbptWPXGJGe0BSy0yoOUOoqTVd/kkIdhS1Qqtpr6fLVQjp3x2Hrb8ohrO84rAo44ui9g/LAZcFl2Wk50l6l6pKajgSDurLGv82FliQTrI9jieUKqvWKNB+P16PRG8T/ydCKdOzd5McX48BB/9B/oOsnevRDiDwApY4Q4dV9vCwcq2klqkQw/CjqLBqKmGKj/j/s2obnOcR0rm5g6swtNmYeTXyWmw4ab/6jp3SlEE17M9R8D/3gbbtbmsfb/pcLhd3XwMXwV775/gzFbxovRUIt0sO77+xivGT7QeP+MZCo0/JaQ/QqLyI6+qiThsjVt4ZWRPmaoanvD/5qQE41aTKoumqBA7CRuryT4IGcIk98ktG6nHaB/7WjuW0SeF0wrDYn6JPGUBvc1DSIA2MFWcTRuq/0DobRpMloNVcsMH0zEqXFbKNIUczQR7LPrMj5McBQv9gyoZGSiTv8bXGWvishny1/GiFxYFi01P71ax9NQNJ4snzp4wJeBB/mllBH7jNZ872RZfg2MCI90mnSJxhADW4/UEGsXWLqQnoqdOqS2RModohCjxgBdyDLUKrbAmPJiOexP7sieRMvVRW1xr1XuEF312mmETKfhMUpM9+vyEFPbWvjYQGDalJjptMW5JCb6SUhqSdliYj3zCsK3hkAMpVIouvhbvgb4W70o9bwYiynC3xYPblsUY/OQlE+bmWada2eBYqmkQ1HTnTUAF+VJxuUvkO1bWLcb9LzTkBIhE7dB8rHIrBDt1XSZWWxDLuHqh3noIFzuyxaHhHZ7wxBioc/HRSp0AA7iXZlsGYchOiMUApmsLY1KEFS/4MPmByf2RydVYL8yfbb623YGoYeLp39OWlT309BlzmoEsGjoilbjZuDdxCsWwxXb6hnv4LzqK8rVHO2oVwHwquXYsM1ZxmR7Jkx+gHNymJ/ob9RcO3xob2tXO7sph2X6ogyFChDcYWA9uIExlLxjrFLk0wHcm/3dRwED3UNRZe+Sbrq5kLuWpndK4G+BAm/8AB9Sz3NaZc9kZLcjtPhTt3sV4eZWnUC/QVD9wam/1jUF/Y2Fv8QWJbK4LuecYfE3EDX9fZ9PPQLt4LGnA3zwBWTNhYna7zomhEbJ7nPSf8trK5o2ZRxYrYMn2Jw0B4aaR5U0qqghurv1zaRDrae6BEYHWV3wYx0oM8D49AhiimewfznSjms+hFewfNejQzL3TYf0dwU7+yJhUGXdA5V/NzlWm+udj/LG7UVml4wuHMipiH9M9Nbg+r/FPVEdxk8oJuil2BJDUSBoFNRplhtsq/0+Z4al170W79jKf/HcwcNVytSUgnDUaaUFqzqCM4S/EWE6LpnVAXOJ/EKQi10vpjOO34uOEQJQgEVsHX1HplQazTuu/pSVN0iAijWIgKIZl84uveQiVMA85wXL7k8MDboQApUZB5jiTTAEkoTc+pcDc2/pHy3Wliv59Lm7BZoh0DBP4sz0CN/RB4Nm4QO8jule4vxNURC/qr3w40GZVEr2PWvtHIXng+YQwDNi/Hkcz/YWaa9YoZppTucV+V0osxc/hWCW9z6/eI5SMzSB1FJh2N6TgJpwd4czcBP6CRhfj1ywWt+KHAO32hzKg+oVNSKLEH41CRADwYM8UTU9BspnVXFFgiphzXhF6a2o4QsmnPsuI4J6u7DImnWQ8D+76gVP/j/j3tFO+MUXZ5/FxZnnmUOdvQ8ZPcb6u3vDm1D25ZxjkO3h6GxpeLyWC9o7JimRFwNJV07+Nc0LXpjDl56vsqZUU2iWVs2hXXHu8LlhT3OB8aOHzwFZakJ9h1BLRq5EFNLVdOzCGttpGtHucFfm7/idx9v0rsT55le/i03euUOryQ+AIQFZjUXR0B4/x7LUlt+G+dkjFXsQ5fBL4EFBj1J5uCj/Pq35Gy985lQPIFO67lUX+CAempu7+rijKVr+LzyRBWdcz9pHc4oYHfqVnDn/nWiCceJQce9uBVqujLcZSO4745wlrfeKJ1r4kMwnFogf/WiZjPxu9M5+exx2BNxEsC2leubA5cKQuS8PgyRyjVjGBlIyGs6bDznvK8jexwzBH1wXH9AqELGUv4rH+rYcB3j0aQ5/03SUUaxagtbxM4sCKlmmbOhvNKCua+7OIIA2MgSxmZ1+6a8j0kHBLhrP0p30rrqE2hnNoNLYnLRuGq7LmVFXdn8cVkle4EokC5AJnzveYNH+316+2UdXg2+0WmBXUTtKALEXAKVF/y7QoM04WTBMVoskGIr3AnKOWdmxrTs52Zmy6Rz9A9w4/cX2XRLacus8H48Sx53dAmOBPHVlBhIAyBE1C7LAzEC/hLNTpMXNY8lexva0anILMK4Tp0g78xCgHaflWUxpzoQ6Tx6jFjMJNROstf9YoF/UguAUXDYG1M3Sc21IPt2QgN58DvABj8qeAN4fV5Qltmfje7S0eq4oy5R0yPB36xX4bGZ6W5SCFqt66caN96kTR61TH73NICxcHhN1DzEmzWfPOhbDeleUqHr2JGJ3pOzVoiQMyjg12OTWPrs2tSxP2vqJj9yC010amIWqB/p6wydOdJ/OXnzjlvT7WBMUFw8HOkik4id9aShVSZUP1cGJmmYm41wOUEsHMSGyVeuPzRN9gxGyGdfV0SrMJFJV6hSyB7r5B91v1lSTs+LMRpTdLGwTL8kth77oQdw0jlCh+h6YRcpw==
```

**Comment by PeterAlfredLee** [comment]
We obtained similar results that `crypto-js-wasm` is slower than `crypto-js` when performing DES decryption, per this result:
!(result)[https://github.com/originjs/crypto-js-wasm/blob/main/benchmark/benchmark_chrome.png]

**Comment by PeterAlfredLee** [comment]
Actually we have discussed this, and we believe this is because:
1. `JavaScript + V8` is not that bad, maybe it's out of JIT?
2. Some algorithms are not that complicated(like AES and DES), in which case the overhead of data copying in WASM can not be iggnored. I think this is an interesting result: maybe we should only use WASM in complicated ciphers/hashers?

Besides this, we have developed a test bench that works in various browsers at [here](https://originjs.org/WASM-benchmark/#/benchmark/browser/collisionDetection). And of course, this benchmark can be run in Node.js. You do not have to reinvent the wheel. :)
